### ISSUE TITLE
Error "'SocketServer' is not defined"

### ISSUE THREAD

**Issue by mythofechelon** [trustedsec/social-engineer-toolkit#686]
**Expected Behaviour**

Built-in SSL server to start

**Actual Behaviour**

With `WEBATTACK_SSL` and `SELF_SIGNED_CERT` set to `ON` (and `APACHE_SERVER` set to `OFF`) in `/etc/setoolkit/set.config`, attempting to start a credential harvest attack using site clone fails with the following output:

```
set:webattack> Enter the url to clone:example.com

[*] Cloning the website: http://example.com                                                                                                                                                                                                
[*] This could take a little bit...                                                                                                                                                                                                        

The best way to use this attack is if username and password form fields are available. Regardless, this captures all POSTs on a website.                                                                                                   
Generating a RSA private key
......................................+++++
.............+++++
writing new private key to 'private/cakey.pem'
Enter PEM pass phrase:
Verifying - Enter PEM pass phrase:
-----
You are about to be asked to enter information that will be incorporated
into your certificate request.
What you are about to enter is what is called a Distinguished Name or a DN.
There are quite a few fields but you can leave some blank
For some fields there will be a default value,
If you enter '.', the field will be left blank.
-----
Organization Name (company) [The Sample Company]:Example
Organizational Unit Name (department, division) []:IT
Email Address []:test@example.com
Locality Name (city, district) [Metropolis]:Example
State or Province Name (full name) [New York]:Example
Country Name (2 letter code) [US]:GB
Common Name (hostname, IP, or your name) []:Example CA
[*] The Social-Engineer Toolkit Credential Harvester Attack
[*] Credential Harvester is running on port 443                                                                                                                                                                                            
[*] Information will be displayed to you as it arrives below:                                                                                                                                                                              
[*] Starting built-in SSL server
[!] Something went wrong.. Printing error: name 'SocketServer' is not defined

The Web Attack module is a unique way of utilizing multiple web-based attacks in order to compromise the intended victim.

The Java Applet Attack method will spoof a Java Certificate and deliver a metasploit based payload. Uses a customized java applet created by Thomas Werth to deliver the payload.

The Metasploit Browser Exploit method will utilize select Metasploit browser exploits through an iframe and deliver a Metasploit payload.

The Credential Harvester method will utilize web cloning of a web- site that has a username and password field and harvest all the information posted to the website.

The TabNabbing method will wait for a user to move to a different tab, then refresh the page to something different.

The Web-Jacking Attack method was introduced by white_sheep, emgent. This method utilizes iframe replacements to make the highlighted URL link to appear legitimate however when clicked a window pops up then is replaced with the malicious link. You can edit the link replacement settings in the set_config if its too slow/fast.

The Multi-Attack method will add a combination of attacks through the web attack menu. For example you can utilize the Java Applet, Metasploit Browser, Credential Harvester/Tabnabbing all at once to see which is successful.

The HTA Attack method will allow you to clone a site and perform powershell injection through HTA files which can be used for Windows-based powershell exploitation through the browser.

   1) Java Applet Attack Method
   2) Metasploit Browser Exploit Method
   3) Credential Harvester Attack Method
   4) Tabnabbing Attack Method
   5) Web Jacking Attack Method
   6) Multi-Attack Web Method
   7) HTA Attack Method

  99) Return to Main Menu

set:webattack>
```

**SET Version**

`8.0.3`

**Kali Version**

```
Description:    Kali GNU/Linux Rolling
Release:        2020.1
Codename:       kali-rolling
````

**Comment by samredinet** [comment]
Seeing the same issue on mine with 
With WEBATTACK_SSL set to ON and SELF_SIGNED_CERT set to OFF
Providing my own *.PEM files

![image](https://user-images.githubusercontent.com/50487351/75725448-64ad6d00-5cd8-11ea-81c8-a2ddf7a733be.png)

**Comment by meitar** [comment]
I have provided patches that fix this issue for Python 3 installs in #696 and #698. Both are required for a complete fix, but these are two different Pull Requests because they are technically different issues, one caused by a Python 2 to 3 migration error (the one described in this issue) and the second caused by an unhandled exception when clients send TLS Alert messages to the underlying OpenSSL libraries that SET uses.
